lib/ocpp/v201/charge_point.cpp

```diff
@@ -1803,6 +1806,33 @@ void ChargePoint::handle_start_transaction_event_response(const EnhancedMessage<
     }
 }
 
+void ChargePoint::handle_get_transaction_status(const Call<GetTransactionStatusRequest> call) {
+    const auto msg = call.msg;
+
+    std::optional<bool> ongoing_indicator;
+    bool messages_in_queue = false;
+
+    if (msg.transactionId.has_value()) {
+        if (this->get_transaction_evseid(msg.transactionId.value()).has_value()) {
+            ongoing_indicator = true;
+        } else {
+            ongoing_indicator = false;
+        }
+        if (this->message_queue->contains_transaction_messages(msg.transactionId.value())) {
+            messages_in_queue = true;
+        }
+    } else if (!this->message_queue->is_transaction_message_queue_empty()) {
+        messages_in_queue = true;
+    }
+
+    GetTransactionStatusResponse response;
```

Comment:
If you moved this to above the if statements you could do the assignments directly ;)